File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.c
# define MAX(a,b) (a>b?a:b)

int maxProfit(int* prices, int pricesSize){
    
    int best=0;
    int temp=0;
    
    for(int i=1;i<pricesSize;i++){
        
        temp+=prices[i]-prices[i-1];
        
        if(temp<0)
            temp=0;

        best=MAX(best,temp);
    }
    
    return best;
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.go
func MAX(a, b int) int {
    if a > b {
        return a
    }else {
        return b
    }
}

func maxProfit(prices []int) int {
    
    var left int = 0
    var right int = 0
    var best int = 0
    
    for ; right < len(prices); {
        
        if prices[left] >= prices[right] {
            left = right
        } else {
            best = MAX(best, prices[right]-prices[left])
        }
        right++
    }
    return best
}